dapps/assets/src/components/digitaltwins/DigitalTwins.ts

```diff
@@ -66,5 +77,17 @@ export default class DigitalTwinsComponent extends mixins(EvanComponent) {
     },
     100,
     { trailing: true }
-  )
-}
\ No newline at end of file
+  );
+
+  async searchHandler () {
+    if (typeof this.search === 'function') {
+      this.search(this.searchTerm);
+    }
+  }
+
+  searchHandlerDebounced = debounce(
```

Comment:
Why not debounce in searchHandler and switch depending on debounce option parameter?